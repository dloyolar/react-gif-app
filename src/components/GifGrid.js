import React, { useState, useEffect } from 'react';

export const GifGrid = ({ category }) => {
  const [count, setCount] = useState(0);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      'https://api.giphy.com/v1/gifs/search?q=one+piece&limit=25&api_key=YzfEcHSohd23s7SuJ3HGLmsMEzUGtXY0';
    const res = await fetch(url);
    const { data } = await res.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    console.log(gifs);
  };

  return (
    <>
      <h3> {category} </h3>
      <h3>{count}</h3>
      <button onClick={() => setCount(count + 1)}></button>
    </>
  );
};
