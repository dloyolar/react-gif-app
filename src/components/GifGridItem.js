import React from 'react';

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className='card animate__animated animate__fadeInDown'>
      <img src={url} alt={title} loading='lazy' />
      <p>{title.toUpperCase() ? title : 'Imagen without title ğŸ˜¢'}</p>
    </div>
  );
};
