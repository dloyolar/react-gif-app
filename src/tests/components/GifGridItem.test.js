import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('test <GifGridItem />', () => {
  const title = 'Titulo';
  const url = 'https://localhost/image.png';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test('should mount <GifGridItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a parragraph with title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should get a img = url and alt props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have animation class', () => {
    const div = wrapper.find('div');
    expect(div.prop('className')).toContain('animate__fadeInDow');
  });
});
