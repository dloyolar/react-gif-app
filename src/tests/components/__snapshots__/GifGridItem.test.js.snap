// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifGridItem /> should mount <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Titulo"
    loading="lazy"
    src="https://localhost/image.png"
  />
  <p>
    Titulo
  </p>
</div>
`;
